controllers/databasecluster_controller.go

```diff
@@ -968,6 +981,27 @@ func (r *DatabaseClusterReconciler) genPXCHAProxySpec(
 	}
 
 	haProxy.PodSpec.Configuration = database.Spec.Proxy.Config
+	if haProxy.PodSpec.Configuration == "" {
+		haProxy.PodSpec.Configuration = haProxyConfigDefault
+	}
+
+	// Ensure there is a env vars secret for HAProxy
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pxcHAProxyEnvSecretName,
```

Comment:
Will we ever need to customize this per DB? Right now we are setting a global secret that will be used by every PXC DBC in a given namespace, which is fine because the value is hardcoded but will we need to customize this is the future?

Reply:
Hmm, good question. I am not sure about that.

If we want to customize this secret per DB, we might possibly also want to customize the HAProxy config file as well. Unless we want the user to configure these, I don't think we will have the need to customize this..

Reply:
Let's keep it like this for now, if we find the need to change it in the future we can do so